### PR TITLE
NIP-09: Add `a` tag deletion

### PR DESCRIPTION
**Pull request by pablof7z**
Simple change to explicitly mention NIP-33 event deletion

### CODE REVIEW

**Comment by arthurfranca** [comment]
Imo it isn't needed. Just fetch the parameterized replaceable event to know its id before sending a deletion event.

**Comment by pablof7z** [comment]
that would just delete the latest event; an `a` tag is to delete the nip-33 itself

**Comment by fiatjaf** [comment]
I think I'll side with @arthurfranca for the first time here.

**Comment by pablof7z** [comment]
imagine deleting an event without `a`:

fetch `"#d": [...], "author"..., "kinds"..`
publish kind:5 with `['e', <latestId>]`
fetch `"#d": [...], "author"..., "kinds"..` + beg to the relay gods that they send you the prior version
publish kind:5 with `['e', <priorId>]`
fetch `"#d": [...], "author"..., "kinds"..` + beg some more
publish kind:5 with `['e', <priorId>]`

this sounds like a nightmare, what if you missed an event? did you just "rebased" the nip-33 to a prior revision?

**Comment by fiatjaf** [comment]
I just heard you saying deletions are a lie on the Miami Bitcoin Conference, @pablof7z.

**Comment by tyiu** [comment]
If you can send a request to replace events with an `a` tag, you should be able to request for deletion of `a` tags. It's the same thing and it should mirror that functionality. I agree with @pablof7z

File: 09.md [hunk 1]
```diff
@@ -20,7 +20,8 @@ For example:
   "pubkey": <32-bytes hex-encoded public key of the event creator>,
   "tags": [
     ["e", "dcd59..464a2"],
-    ["e", "968c5..ad7a4"]
+    ["e", "968c5..ad7a4"],
+    ["a", "968c5..ad7a4"]
```
Comment (tyiu) [inline 09.md:24]: Instead of a hex key as `e` has it, `a` should be in the format of

```
["a", "<kind>:<pubkey>:<d-identifier>", "<relay url>"]
```

**Comment by fiatjaf** [comment]
I agree. This should be good then. The only problem is that it creates an extra requirement for relays and confusion on clients about how to properly delete things.

**Comment by pablof7z** [comment]
precisely, given that it would be weird to store all versions, but you can't assume they don't either; the safe approach would be deleting NIP-33s via `a`.

**Comment by arthurfranca** [comment]
xd I wasn't considering this scenario. Your point is valid.
Just need to remove the  "<relay url>" part that makes no sense as you are already sending de deletion event to a specific relay.

File: 09.md [hunk 2]
```diff
@@ -20,7 +20,8 @@ For example:
   "pubkey": <32-bytes hex-encoded public key of the event creator>,
   "tags": [
     ["e", "dcd59..464a2"],
-    ["e", "968c5..ad7a4"]
+    ["e", "968c5..ad7a4"],
+    ["a", "<kind>:<pubkey>:<d-identifier>", "<relay url>"]
```
Comment (arthurfranca) [inline 09.md:24]:
```suggestion
    ["a", "<kind>:<pubkey>:<d-identifier>"]
```
small fix

**Pull request merged**